qt/python/mantidqt/widgets/fitpropertybrowser/addfunctiondialog/view.py

```diff
@@ -24,13 +26,34 @@ def _setup_ui(self, function_names):
         self.ui = load_ui(__file__, 'add_function_dialog.ui', self)
         if function_names:
             self.ui.functionBox.addItems(function_names)
+        self.ui.functionBox.clearEditText()
         self.ui.functionBox.completer().setCompletionMode(QCompleter.PopupCompletion)
         self.ui.functionBox.completer().setFilterMode(Qt.MatchContains)
         self.ui.errorMessage.hide()
+        # Monkey patch the functionBox keyPressEvent to make use of auto-completion.
+        self.ui.functionBox.keyPressEvent = types.MethodType(keyPressEvent,
+                                                             self.ui.functionBox)
 
     def is_text_in_function_list(self, function):
         return self.ui.functionBox.findText(function, Qt.MatchExactly) != -1
 
     def set_error_message(self, text):
         self.ui.errorMessage.setText("<span style='color:red'> %s </span>" % text)
         self.ui.errorMessage.show()
+
+
+# This method monkey patches the keyPressEvent for the QComboBox function box
+# Using this patch, the text will be auto-completed if enter is pressed
+# and the number of possible completions is one.
+def keyPressEvent(self, event):
+    if event.key() == Qt.Key_Return:
+        # if completion list is of len = 1, accept the completion else continue
+        if self.completer().completionCount() == 1:
+            completion_text = self.completer().currentCompletion()
+            if completion_text != self.currentText():  # manually emit the code completion signal
+                self.completer().activated.emit(completion_text)
+    # If up or down pressed, start the completer
+    if event.key() in [Qt.Key_Down, Qt.Key_Up]:
```

Comment:
You could do `elif` instead of `if` so the check isn't carried out if the enter key is pressed.

Reply:
Good spot